### PR TITLE
add button alignment

### LINKED ISSUE #2253: Button - Add "button ordering" guidance on site
(From #2242)

Add documentation/ examples from [Figma](https://www.figma.com/file/Oi5UEEQ33VCAyOKQcZ8Nr8/HPE-Button-Component?node-id=5237%3A9596) to [Guidance ](https://design-system.hpe.design/components/button?q=butto#guidance)section, after "Button Alignment"

### PR DESCRIPTION
**Pull request by britt6612**
[Deploy Preview](https://deploy-preview-2302--keen-mayer-a86c8b.netlify.app/components/button#button-ordering)

#### What does this PR do?
This PR adds the `alignment` examples from Figma to the site for guidance around placement of Buttons 
#### Where should the reviewer start?
button.mdx
#### What testing has been done on this PR?

In addition to the feature you are implementing, have you checked the following:

**General UX Checks**
- [x] Small, medium, and large screen sizes
- [x] Cross-browsers (FireFox, Chrome, and Safari)
- [x] Light & dark modes
- [ ] All hyperlinks route properly

**Accessibility Checks**
- [x] Keyboard interactions
- [ ] Screen reader experience
- [x] Run WAVE accessibility plugin (Chrome)

**Code Quality Checks**
- [x] Console is free of warnings and errors
- [x] Passes E2E commit checks
- [ ] Visual snapshots are reasonable

#### What are the relevant issues?
closes #2253 
#### Should this PR be mentioned in Design System updates?
yes
#### Is this change backwards compatible or is it a breaking change?
 backwards compatible

### CODE REVIEW

File: aries-site/src/pages/components/button.mdx [hunk 1]
```diff
@@ -56,6 +58,19 @@ Buttons are unique components since their alignment depends on where they appear
 
 <ButtonAlignmentTable />
 
+### Button Ordering
```
Comment (taysea) [inline aries-site/src/pages/components/button.mdx:61]:
```suggestion
### Button Ordering

```

**Comment by taysea** [comment]
The examples don't align with what is shown on [Figma](https://www.figma.com/file/Oi5UEEQ33VCAyOKQcZ8Nr8/HPE-Button-Component?node-id=5237%3A9596)

Figma screenshot (right-aligned should be default, secondary, primary):
Screen Shot 2022-02-01 at 12 19 31 PM

File: aries-site/src/pages/components/button.mdx [hunk 2]
```diff
@@ -56,6 +58,20 @@ Buttons are unique components since their alignment depends on where they appear
...
+
+The Primary button always comes first, followed by any Secondary buttons, then by any Default buttons.
```
Comment (halocline) [inline aries-site/src/pages/components/button.mdx:63]: I like the succinctness of this sentence, however I am thinking of cases where a Primary button is not present, and speaking to the right-aligned scenario.

```suggestion
Button types may be applied in various combinations, however they should alway be ordered by their level of importance: Primary first, followed by Secondary, then by Default.

The order direction is determined by the button's alignment. Left-aligned buttons should be ordered from left to right; right-aligned, from right to left.
```

@CitizenFox, any suggestions?

**Review by halocline — APPROVED** [review]
Code looks good. 

Tagging @CitizenFox for any copy thoughts.

**Review by taysea — APPROVED** [review]
Code looks good, let's give a little time for Will to take a look at the copy if he has time. But if he doesn't we can move forward.

**Comment by halocline** [comment]
> Code looks good, let's give a little time for Will to take a look at the copy if he has time. But if he doesn't we can move forward.

@CitizenFox is coming!

File: aries-site/src/pages/components/button.mdx (hunk 2 above)
Comment (CitizenFox) [reply aries-site/src/pages/components/button.mdx:63]:
```suggestion
Button types may be applied in various combinations, however they should always be ordered by order of importance and in relation to their alignment: Primary first, followed by Secondary, then by Default.
```
I like your suggestion, @halocline. I added a few words to tie order to alignment. Does this make sense?

**Review by CitizenFox — APPROVED** [review]
I left one comment, tagging onto @halocline's. It adds more text to the sentence but I think it's important to capture that order depends on priority in relation to left- or right-alignment. Otherwise, looks good!

**Pull request merged**